### PR TITLE
fix: mediaflux checksum

### PR DESCRIPTION
**Pull request by kcho**
We expected `unimelb-mf-download` to check the checksum of the local data compared to the source data on Mediaflux. But when a file exist in the same file path given to the `--out` argument to the `unimelb-mf-download`, it completely skips downloading the file regardless of the content of the file.

The following lines reproduces the bug
```sh
# create an empty file
touch test.zip

unimelb-mf-download \
    --mf.config mflux.cfg \
    --out ./ \
    --csum-check \
    /projects/proj-5070_prescient-1128.4.380/PrescientXX/test.zip
```

Returns
```sh
Skipped asset 1264700582: '/projects/proj-5070_prescient-1128.4.380/PrescientXX/test.zip'. Already exists.
```

This PR extracts, saves, and uses the CRC32 checksum from the `unimelb-mf-check` to re-download changed files from Mediaflux.

### CODE REVIEW

File: lochness/mediaflux/__init__.py [hunk 1]
```diff
@@ -163,6 +163,18 @@ def sync_module(Lochness: 'lochness.config',
                                     if 'out_dir' in prod else mf_local) / \
                                         subpath.parent
 
+                            # do not re-download already downloaded data
+                            prev_checksum_file = Path(mf_local).parent / \
+                                    f'.checksum_{Path(mf_local).name}'
+                            if prev_checksum_file.is_file():
```
Comment (tashrifbillah) [inline lochness/mediaflux/__init__.py:169]: For each file, are you maintaining a separate checksum file?

File: lochness/mediaflux/__init__.py [hunk 2]
```diff
@@ -184,13 +196,9 @@ def sync_module(Lochness: 'lochness.config',
                                       stdout=DEVNULL, stderr=STDOUT)
                             p.wait()
 
-                            # verify checksum after download completes if
-                            # checksum does not match, data will be downloaded
-                            # again ENH should we verify checksum 5 times?
-                            cmd += ' --csum-check'
-                            p = Popen(cmd, shell=True,
-                                      stdout=DEVNULL, stderr=STDOUT)
-                            p.wait()
```
Comment (tashrifbillah) [inline lochness/mediaflux/__init__.py:193]: Does this block not work that you had to write the new code? This block wasn't re-downloading data. It was first calculating checksum of downloaded data to decide if re-download would be necessary. It appears that you are doing  the same thing using information from your `diff_path`.

Or is is that your `diff_path` already contains checksum so you do not want to redo it by this block?

File: lochness/mediaflux/__init__.py (hunk 1 above)
Comment (kcho) [reply lochness/mediaflux/__init__.py:169]: Yes, that's the idea.

File: lochness/mediaflux/__init__.py (hunk 2 above)
Comment (kcho) [reply lochness/mediaflux/__init__.py:193]: `diff_path` contains the checksum of the source file in the `SRC_CHECKSUM` column, which this PR is trying to use in deciding when to re-download already downloaded files.

**Review by tashrifbillah — APPROVED** [review]

**Pull request merged**